R/geom-col.r

```diff
@@ -42,16 +44,19 @@ GeomCol <- ggproto("GeomCol", GeomRect,
     params
   },
 
-  extra_params = c("na.rm", "orientation"),
+  extra_params = c("just", "na.rm", "orientation"),
 
   setup_data = function(data, params) {
     data$flipped_aes <- params$flipped_aes
     data <- flip_data(data, params$flipped_aes)
     data$width <- data$width %||%
       params$width %||% (resolution(data$x, FALSE) * 0.9)
-    data <- transform(data,
-                      ymin = pmin(y, 0), ymax = pmax(y, 0),
-                      xmin = x - width / 2, xmax = x + width / 2, width = NULL
+    data$just <- data$just %||% params$just %||% 0.5
+    data <- transform(
+      data,
+      ymin = pmin(y, 0), ymax = pmax(y, 0),
+      xmin = x - width * (1 - just), xmax = x + width * just,
+      width = NULL, just = NULL
```

Comment:
This `just = NULL` as well.

Reply:
Sorry, I'm not quite sure what approach you're advocating... Currently we have this:
```
    data$just <- data$just %||% params$just %||% 0.5
    data <- transform(
      data,
      ymin = pmin(y, 0), ymax = pmax(y, 0),
      xmin = x - width * (1 - just), xmax = x + width * just,
      width = NULL, just = NULL
    )
```
Are you saying we shouldn't remove the `just` column? In that case we'd have this, but that seems a bit odd as `width` gets removed:
```
    data$just <- data$just %||% params$just %||% 0.5
    data <- transform(
      data,
      ymin = pmin(y, 0), ymax = pmax(y, 0),
      xmin = x - width * (1 - just), xmax = x + width * just,
      width = NULL
    )
```

Do you think the whole `setup_data` field for `geom_col()` and `geom_bar()` should have its own named function (e.g. `setup_col_data()`) to avoid code duplication btw? For my money the logic is complex enough that it'd be quite a bit safer to have it in one place.

Reply:
Really sorry, for some reason, I simply didn't read this line... You are right. Please forget my comment above.

```r
data$just <- data$just %||% params$just %||% 0.5
```

Reply:
Haha, I did wonder! No worries at all 😄